sonaranalyzer-dotnet/src/SonarAnalyzer.CFG/ControlFlowGraph/CSharpControlFlowGraphBuilder.cs

```diff
@@ -815,7 +815,10 @@ private Block BuildSwitchExpression(SwitchExpressionSyntaxWrapper switchExpressi
 
                 currentArmBlock = BuildArmBranch(arm, armBlock, currentArmBlock, isLast);
 
-                currentArmBlock = BuildExpression(switchExpressionSyntax.GoverningExpression, currentArmBlock);
+                if (!isLast)
+                {
+                    currentArmBlock = BuildExpression(switchExpressionSyntax.GoverningExpression, currentArmBlock);
+                }
```

Comment:
Could we extract the last arm outside the loop ? That would avoid the need for the `isLast` boolean completely and simplify the loop. WDYT ?

Reply:
If I understood correctly I don't think that that it's possible since we want to add the expression for all the arms but the last.

Reply:
It could be possible, but indeed, I m not sure we would win much readability actually, so forget about my comment